### PR TITLE
Add props to Object table to permit a label in the fab button

### PR DESCRIPTION
**Pull request by matiasgarcia91**
image

### CODE REVIEW

**Comment by SferaDev** [comment]
If we follow the approach from Metadata Sync, we decided to customize the button in the app instead of the ui library.

**Review by tokland — APPROVED** [review]
Tested with vaccination-app, works as expected.

**Comment by tokland** [comment]
Prop `createButtonLabel` changed to `PropTypes.node`, so it can be anything renderable (typically: string or React element):

https://reactjs.org/docs/typechecking-with-proptypes.html#proptypes

@adrianq, @SferaDev  Ready for review. Once merged, I'll update to npmjs so we can use it in metadata-sync.

**Review by SferaDev — APPROVED** [review]
Tested on metadata-sync

image

image

File: src/objects-table/ObjectsTable.js [hunk 1]
```diff
@@ -421,7 +430,7 @@ class ObjectsTable extends React.Component {
                     ) : null}
                 </div>
 
-                {onCreate && <ListActionBar onClick={onCreate} />}
+                {onCreate && <ListActionBar onClick={onCreate} label={createButtonLabel} />}
```
Comment (SferaDev) [inline src/objects-table/ObjectsTable.js]: *Minor:* I know it's a breaking change (should be avoided if third parties used the library) but wouldn't it make sense something like:

```
{floatingButtonAction && <ListActionBar onClick={floatingButtonAction} label={floatingButtonLabel} />}
```

I understand it originally meant "Create" but now it's generic enough to be called something else.

**Review by SferaDev — APPROVED** [review]

**Comment by adrianq** [comment]
@tokland Works fine. I have just changed the prop to buttonLabel as suggested by @SferaDev. I have already taken care of propagating this change to vaccination repo. @SferaDev can you do the same with metadata sync? @tokland Whenever you have a change if you can release to npm...

**Comment by SferaDev** [comment]
@adrianq Updated accordingly https://github.com/EyeSeeTea/metadata-synchronization/pull/92

**Pull request merged**